debian/matrix-synapse-py3.postinst

```diff
@@ -40,12 +41,12 @@ EOF
     /opt/venvs/matrix-synapse/lib/manage_debconf.pl update
 
     if ! getent passwd $USER >/dev/null; then
-      adduser --quiet --system --no-create-home --home /var/lib/matrix-synapse $USER
+      adduser --quiet --system --group --no-create-home --home /var/lib/matrix-synapse $USER
     fi
 
     for DIR in /var/lib/matrix-synapse /var/log/matrix-synapse /etc/matrix-synapse; do
       if ! dpkg-statoverride --list --quiet $DIR >/dev/null; then
-        dpkg-statoverride --force --quiet --update --add $USER nogroup 0755 $DIR
+        dpkg-statoverride --force --quiet --update --add $USER $GROUP 0755 $DIR
```

Comment:
Will this fall over if somebody upgrades from a previous version?

If I'm right, the group won't be automatically created because `getent passwd $USER >/dev/null` will be true.

I don't know about `dpkg-statoverride` though.

Reply:
I've been running with `matrix-synapse` having the associated `matrix-synapse` group for multiple years now, so e.g. `/var/lib/matrix-synapse` is owned by `matrix-synapse:matrix-synapse`. 

I just ran
```console
# dpkg-statoverride --list /var/lib/matrix-synapse
matrix-synapse nogroup 755 /var/lib/matrix-synapse
```
and it had an exit code of 0, so one doesn't seem to affect the other. That means, this will not fail on update, but it will also not change anything upon update and only affect new installations. 

I've only tested this by manually running the commands. I can't rule out it's affected by some secret environment switches during install, but I think it should be safe.

Reply:
It shouldn't break on upgrade, because there will already be a `dpkg-statoverride` in place so the condition at line 48 will fail.

More relevant is what happens if there is already a `matrix-synapse` user (possibly from a previous installation), and a new version of the package is installed.

Reply:
If we replace `$GROUP` with `$(id -gn $USER)`, then I think this will do the right thing whether or not the `matrix-synapse` group exists.

Reply:
Changed accordingly.